assignment10/get_books.py

```
@@ -0,0 +1,79 @@
+from selenium import webdriver
+from selenium.webdriver.chrome.service import Service as ChromeService
+from webdriver_manager.chrome import ChromeDriverManager
+from selenium.webdriver.common.by import By
+import pandas as pd
+import json
+import time
+import os
+
+
+# Start the driver
+driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()))
+
+url = "https://durhamcounty.bibliocommons.com/v2/search?query=learning%20spanish&searchType=smart"
+driver.get(url)
+time.sleep(5)
+
+
+# Find all book entries
+entries = driver.find_elements(By.CSS_SELECTOR, "li.cp-search-result-item")
+print( len(entries), "entries")
+
+results = []
+
+for entry in entries:
+    # Title
+    try:
```

Instead of catching all exceptions silently, it’s better to catch specific ones or at least print an error when scraping fails. Right now, if something changes on the site, it will just store "N/A" without warning. Just thought to bring your attention to this.